### ISSUE TITLE
The GPU utilization on AutoDL is not high, around 4%.

### ISSUE THREAD

**Issue by gl443** [pytorch/pytorch#132478]
### 🐛 Describe the bug

import pandas as pd

data  =pd.read_csv("mlbootcamp5_train.csv")
x = data[['height', 'weight']] #(70000, 2)
y = data['gender'] # (70000,)
y = y.apply(lambda gender: 1 if gender == 'M' else 0)

import torch
import torch.nn as nn
import  torch.optim as optim
import matplotlib.pyplot as plt
from torch.utils.data import DataLoader, TensorDataset

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')

class NearalNetwork(nn.Module):
    def __init__(self):
        super(NearalNetwork, self).__init__()
        self.fc1 = nn.Linear(2, 2)
        self.fc2 = nn.Linear(2, 1)
        self.sigmoid = nn.Sigmoid()

    def forward(self, x):
        x = self.fc1(x)
        x = self.sigmoid(x)
        x = self.fc2(x)
        x = self.sigmoid(x)
        return x

learning_rate = 0.001
epochs = 1000
losses = []

data1 = torch.tensor(x.values, dtype=torch.float32).to(device)
all_y_lables = torch.tensor(y.values, dtype=torch.float32).unsqueeze(1).to(device)

datasets = TensorDataset(data1, all_y_lables)
dataloader = DataLoader(datasets, batch_size=64, shuffle=True,num_workers=0)

model = NearalNetwork().to(device)
criterion = nn.BCELoss()
optimizer = optim.SGD(model.parameters(), lr=learning_rate)

def trian(data, all_y_lables):
    global epochs # 声明全局变量
    for epochs in range(epochs):
        for batch_data, batch_labels in dataloader:
            batch_data,batch_labels = batch_data.to(device), batch_labels.to(device)
            model.train()
            optimizer.zero_grad()

            y_pred = model(batch_data)

            loss = criterion(y_pred, batch_labels)
            loss.backward()
            optimizer.step()

        if epochs % 20 == 0:
            print(f'epochs: {epochs}, loss: {loss.item()}')
            losses.append(loss.item())

trian(data, all_y_lables)

plt.plot(range(0, epochs, 20), losses)
plt.title("loss over epochs")
plt.ylabel('loss')
plt.xlabel('epoch')
plt.grid(True)
plt.show()

# torch.save(model.state_dict(), 'model.pth')

model.eval()
with torch.no_grad():
    predictions = model(data1).cpu().detach().numpy()
print("Predictions:", predictions)

### Versions

![image](https://github.com/user-attachments/assets/e83c556a-523a-498e-972d-9d04bbc7e89e)

**Comment by tringwald** [comment]
Your network only contains 2 very small matmuls, which will never saturate any kind of GPU. This is not a bug in PyTorch. For general questions, please use [the PyTorch forums](https://discuss.pytorch.org/).
